src/main/java/bio/terra/service/filedata/google/firestore/FireStoreFileDao.java

```diff
@@ -177,20 +184,20 @@ void deleteFilesFromDataset(Firestore firestore, String datasetId, Interruptible
             collectionId,
             DELETE_BATCH_SIZE,
             document -> {
-                FireStoreFile fireStoreFile = document.toObject(FireStoreFile.class);
-                func.accept(fireStoreFile);
-                doDelete(document);
+                SettableApiFuture<WriteResult> future = SettableApiFuture.create();
+                executor.execute(() -> {
+                    try {
+                        FireStoreFile fireStoreFile = document.toObject(FireStoreFile.class);
+                        func.accept(fireStoreFile);
+                        future.set(document.getReference().delete().get());
```

Comment:
is the .get() after the .delete() possible because it's a future?

Reply:
Haha, no worries...this is admittedly a little weird because I'm mixing GCS and Firestore interaction here.  `.delete` kicks off the delete and `.get` waits for it to finish

Reply:
oooooohhhhhh interesting